### PR TITLE
Change methode afect default warehouse

### PR DESCRIPTION
**Pull request by BB2A-Anthony**
# Bug [*Change methode afect default warehouse*]

If an order was generated from a proposal the warehouse was not assigned correctly. If a warehouse is defined on the user, this was not taken into account.
Because we focus on the select.
This will change all other selects. When creating or modifying a warehouse, add or modify in the "Add in" field by the warehouse present on the user what could lead to errors or unintentional modifications!

### CODE REVIEW

File: htdocs/commande/class/commande.class.php [hunk 1]
```diff
@@ -879,6 +879,11 @@ public function create($user, $notrigger = 0)
 		// $date_commande is deprecated
 		$date = ($this->date_commande ? $this->date_commande : $this->date);
 
+		if (empty($this->warehouse_id) || $this->warehouse_id  = -1){
```
Comment (eldy) [inline htdocs/commande/class/commande.class.php:882]: $this->warehouse_id  = -1  is surely a bug. == ?

Comment (bb2a) [reply htdocs/commande/class/commande.class.php]: Yes is not bug i change

**Comment by bb2a** [comment]
@eldy it's resolved no?

**Comment by BB2A-Anthony** [comment]
@eldy this PR fix one bug to warehouse afect for v13

File: htdocs/commande/class/commande.class.php (hunk 1 above)
Comment (bb2a) [reply htdocs/commande/class/commande.class.php]: @eldy Sory an error please valid in V13

**Comment by eldy** [comment]
"Default warehouse" means the value is preselected on screen with this value.
Here, you make a change that will force the warehouse by the low level create method. This means if we ask to create with no warehouse defined (for example using the API), the creation will change the warehouse and will not respect the input value provided by API.

Also, i am still not able to reproduce the bug #14627 mentioned by this PR.
I reproduced the step on the video and i got a different result. Problem does not appears.

**Comment by BB2A-Anthony** [comment]
> "Default warehouse" means the value is preselected on screen with this value.
> Here, you make a change that will force the warehouse by the low level create method. This means if we ask to create with no warehouse defined (for example using the API), the creation will change the warehouse and will not respect the input value provided by API.
> 
> Also, i am still not able to reproduce the bug #14627 mentioned by this PR.
> I reproduced the step on the video and i got a different result. Problem does not appears.

Bug if the order is created by the workflow from a propal the warehouse is not affected. Not a vidéo 

if the option is enabled, this is also the goal

**Comment by eldy** [comment]
If the option is set, when creating the proposal, the default warehouse should have been set in the combo list. If the warehouse is not saved into the proposal, it means, the user has changed the preselected value to force to an unknown value.
So if user would like to force to unknown, when the order is created automatically by the workflow, we must keep the choice of user to unknown, won't it ?

**Comment by BB2A-Anthony** [comment]
> If the warehouse i



> If the option is set, when creating the proposal, the default warehouse should have been set in the combo list. If the warehouse is not saved into the proposal, it means, the user has changed the preselected value to force to an unknown value.
> So if user would like to force to unknown, when the order is created automatically by the workflow, we must keep the choice of user to unknown, won't it ?

In proposal is  not posible to affect warehouse!

If posible is solution. the goal is that an order is strongly assigned to a warehouse.

**Comment by eldy** [comment]
Sorry, i still don't understand what is the bug. Can you describe steps you do to reproduce the bug ?

**Comment by BB2A-Anthony** [comment]
The video
[https://youtu.be/PEVGCvrTByk](https://youtu.be/PEVGCvrTByk)

**Pull request closed without merging**